python/cudf/cudf/core/column/column.py

```diff
@@ -70,42 +71,34 @@ def as_frame(self):
     @property
     def data_array_view(self):
         """
-        View the data as a device array or nvstrings object
+        View the data as a CuPy ndarray or nvstrings object
         """
         if self.dtype == "object":
             return self.nvstrings
 
         if is_categorical_dtype(self.dtype):
             return self.codes.data_array_view
-        else:
-            dtype = self.dtype
 
-        result = rmm.device_array_from_ptr(
-            ptr=self.data.ptr, nelem=len(self), dtype=dtype
-        )
-        result.gpu_data._obj = self
+        result = cp.asarray(self.data).view(self.dtype)
         return result
 
     @property
     def mask_array_view(self):
         """
-        View the mask as a device array
+        View the mask as a CuPy ndarray
         """
-        result = rmm.device_array_from_ptr(
-            ptr=self.mask.ptr, nelem=self.mask.size, dtype=np.int8,
-        )
-        result.gpu_data._obj = self
+        result = cp.asarray(self.mask)
         return result
 
     def __len__(self):
         return self.size
 
     def to_pandas(self):
         arr = self.data_array_view
-        sr = pd.Series(arr.copy_to_host())
+        sr = pd.Series(cp.asnumpy(arr))
 
         if self.nullable:
-            mask_bits = self.mask_array_view.copy_to_host()
+            mask_bits = cp.asnumpy(self.mask_array_view)
```

Comment:
This looks like it's fed into a numba kernel immediately below and should probably just be a device array

Reply:
Good catch! Yeah that makes sense. Wonder why this had `.copy_to_host()` then?

Reply:
Likely just an oversight because we are eventually feeding the result into a Pandas masking call, but there's an explicit copy to host for that.

Reply:
Broke out a smaller fix for this so we can get it in sooner. ( https://github.com/rapidsai/cudf/pull/4092 )